corporate/lib/stripe.py

```diff
@@ -3335,9 +3334,17 @@ def invoice_plan(self, plan: CustomerPlan, event_time: datetime) -> None:
                 stripe_invoice = stripe.Invoice.create(**invoice_params)
                 stripe.Invoice.finalize_invoice(stripe_invoice)
 
+        plan.invoicing_status = CustomerPlan.INVOICING_STATUS_DONE
         plan.next_invoice_date = next_invoice_date(plan)
         plan.invoice_overdue_email_sent = False
-        plan.save(update_fields=["next_invoice_date", "invoice_overdue_email_sent"])
+        plan.save(
+            update_fields=[
+                "next_invoice_date",
+                "invoice_overdue_email_sent",
+                "invoicing_status",
+                "invoiced_through",
+            ]
+        )
```

Comment:
Deferring the database update to the end of the loop seems dangerous for error-handling. If one of the later loop iterations fails, won't that leave us with a potentially larger divergence of state between Zulip and Stripe?

Reply:
Sorry, forgot to address this. Opened #34643 to fix.